### DISCUSSION TITLE
Improvements for the osu!(lazer) editor

### DISCUSSION THREAD

**Discussion by Tenexxt** [ppy/osu#29014] [Ideas]
**Overall Editor**

- **"Copy timings to all difficulties"** - Would be usefull with mapsets with multiple difficulties, especialy with the ones that have over 10. As the name sugests, this option would copy timings to all difficulties of the set. It could be found in the same panel as "Set preview point to current time"
- **"Copy metadata to all difficulties"** - copying it to all difficulties one by one, especially on sets with a lot of them, can be annoying. Adding this would be usefull and could save mappers some time, not necessarily a lot, but its still something. Could be placed in the setup chanel of the editor, under the metadata section.
- **Drain time shown on the bottom bar** - isn't something really needed, but could be usefull when you didn't map the entire song, and just want to compare it to the normal size of the song, or when you just want to know when you are actually playing in the song.
- **If timing between notes is to small, display it in the verify chanel** - Could be usefull for easier difficulties. The game for an example, when detects a 1/4 note, when on a certain difficulty they arent supposed to be on, display it as a warning.
- **If metadata isnt matching in all difficulties, display it in the verify chanel** - self explainatory. Since osu web picks metadata from one of the difficulties, it could save the confusion why certain tags aren't displayed on the web.
- **Visual changes on the timeline** - BPM in the latest lazer ver, is a super similar, if not the same colour as the 1/2 beat, which is just a little annoying to me, id be glad if it was changed. Its more of a personal opinion, so this can be ignored.
- **Mod testing in the editor** - some maps are better with mods on, or mappers just want to optimize the map for those mods. A way to add mod select to testing would be, that under the test button you could roll out the mod window.

**osu!taiko Editor**

- **Visual changes to notes on the timeline** - normal hits dont have any visual difference on the timeline from the strong ones, same for sliders (drumroll) and spiners (Swell). Making Strong hits bigger on the timeline isn't a good idea also, as it could cover some important visuals, so instead of making them bigger, they can have some smaller highlight inside of them, like this: https://drive.google.com/file/d/1rVx-KMiXMc2w0pzn7FaUpDmpp3_4PZGN/view?usp=sharing As for sliders and spiners, just make one of them yellow (since they are both yellow, but they have to be distinct on the timeline)
- **Visual changes to the timeline** - unlike osu!standard and osu!catch, the timeline doesn't need as much space as it gets. it requires less space, because slider speed can't be adjusted, so there cant be anything on the upper part of the timeline, so making it smaller would be nice.
- **Object names not having spaces in-between words** - DrumRoll instead of Drum Roll. Add spaces in-between

**osu!catch Editor**

- **Grid** - Adding grid options similar to osu!standard, espacially circle and triangle ones could help with more accurate pattern-making.
- **Object names not having spaces in-between words** - Juice Stream is called JuiceStream, Banana Shower is BananaShower. Add spaces in-between

**osu!mania Editor**

- **Visual changes to notes on the timeline** - Since on the argon skin on lazer different keys have their own colour, making the notes on that key the colour of it. This is a small thing, that i doubt will help many, but it will make more sense than when looking at the timeline all you see are cyan notes that have no difference in-between them. If this idea is bad, **then atleast make more visual difference between notes and sliders.**

**osu! Editor**

- **Place a button nearby the grid settings that resets its values** - Having to go out of the editor just to reset them is kinda annoying, so place a button nerby them that resets the X and Y offset, spacing and rotation - of course, every value has a seperate one

Don't have anything more to say, the editor is fine to me so i don't have many issues with it.

Copied over from my post on reddit: https://www.reddit.com/r/osugame/comments/1e8v6n3/my_ideas_for_the_lazer_editor/
Edit: added a another idea in the Overall Editor section

**Comment by WangleLine** [comment]
"Copy timings to all difficulties" and "Copy metadata to all difficulties" would be such a timesaver.
Been a little annoyed by the repetition in updating tags across diffs
